application/utils/harvester/diff_retriever.py

```
@@ -0,0 +1,99 @@
+        diff_size = len(diff_bytes)
+
+        if diff_size > self.MAX_DIFF_SIZE_BYTES:
+            raise ValueError(
+                f"Diff size ({diff_size} bytes) exceeds "
+                f"maximum supported size ({self.MAX_DIFF_SIZE_BYTES} bytes)."
+            )
+
+        return diff_bytes.decode("utf-8", errors="replace")
+
+    def _resolve_commit(self, commit: str) -> str:
+        result = subprocess.run(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)